ccbuilder/builder/builder.py

```diff
@@ -413,58 +373,35 @@ def build(
             self:
             project (CompilerProject): Project to build.
             rev (str): Revision of the project to build.
-            get_executable (bool): Whether or not to return the path to the executable compiler instead.
             additional_patches (Optional[list[Path]]): Additional patches to apply.
             jobs (Optional[int]): Amount of jobs to use for the building.
             configure_flags (str | None): additional flags to pass to the configure script or cmake.
             force (bool): Build the specified compiler even if it has been cached before.
 
         Returns:
-            Path: `cache_prefix`/projectname-commit or `cache_prefix`/projectname-commit/bin/$compiler if `get_executable` is set.
+            Path: `cache_prefix`/projectname-commit
         """
-
         repo = select_repo(project, self.llvm_repo, self.gcc_repo)
-        commit = repo.rev_to_commit(rev)
-        install_path = get_install_path(self.cache_prefix, project, commit)
-        success_indicator = install_path / "DONE"
-        worker_indicator = install_path / "WORKER_PID"
-
-        postfix: Path = get_executable_postfix(project) if get_executable else Path()
-
-        if not force:
-            if success_indicator.exists():
-                return install_path / postfix
-            elif install_path.exists() and _worker_alive(worker_indicator):
-                logging.info(
-                    f"This compiler seems to be built by some other worker. Need to wait. If there is no other worker, abort this command and run ccbuilder cache clean."
-                )
-                start_time = time.time()
-                counter = 0
-                while (
-                    _worker_alive(worker_indicator)
-                    and not success_indicator.exists()
-                    and install_path.exists()
-                ):
-                    time.sleep(1)
-                    if time.time() - start_time > 15 * 60:
-                        counter += 1
-                        logging.info(
-                            f"{counter*15} minutes have passed waiting. Maybe the cache is in an inconsistent state."
-                        )
-                        start_time = time.time()
-
-                if success_indicator.exists():
-                    return install_path / postfix
-
-                # Someone was building but did not leave the success_indicator
-                # so something went wrong with the build.
-                raise BuildException(f"Other build attempt failed for {install_path}")
-
-        return super().build(
-            project,
-            rev,
-            get_executable=get_executable,
```

Comment:
Two things:  `get_executable` is still present in the method parameters although you don't want it anymore.

The get executable was there because if I want the executable and I'm given the `CompilerProject`, I don't want to write a `match project:` every time.
I suggest either getting the argument back or having an additional function which can add the postfix.

Reply:
What I don't like with the `get_executable` is that it returns the C wrapper (e.g., `gcc`) but not the C++ one (e.g., `g++`). I think a better solution is to return something like compilerstore.BuiltCompilerInfo and use it's `.to_compiler() ->CompilerExe` method (I'll include this)